packages/terra-navigation-prompt/src/NavigationPrompt.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import uuidv4 from 'uuid/v4';
+
+import withPromptRegistration from './_withPromptRegistration';
+
+const propTypes = {
+  /**
+   * A string describing the content or concept for which the NavigationPrompt is being rendered.
+   */
+  description: PropTypes.string,
+  /**
+   * An object containing any other pertinent information related to the NavigationPrompt.
+   */
+  metaData: PropTypes.object,
+  /**
+   * @private
+   * An object containing prompt registration APIs provided through the PromptRegistrationContext.
+   */
+  promptRegistration: PropTypes.shape({
+    isDefaultContextValue: PropTypes.bool,
```

Comment:
Add some documentation about what this means maybe?

Reply:
Should we export this shape from `PromptRegistrationContext` ?

Reply:
Yes and yes